File: combinatorics/resource/ST/project_5/bai_toan_4/simple_graph/al_to_amap.cpp
#include <iostream>
#include <vector>
#include <map>
#include <sstream>

using namespace std;

// Ham chuyen danh sach ke sang ban do ke
map<int, vector<int>> adjacencyListToMap(const vector<vector<int>>& adjList, int n) {
    map<int, vector<int>> adjMap;  // Khoi tao ban do ke

    for (int i = 0; i < n; ++i) {
        adjMap[i] = adjList[i];  // Gan danh sach ke cua i vao map
    }

    return adjMap;
}

int main() {
    int n;
    cout << "Nhap so dinh: ";
    cin >> n;
    cin.ignore();

    vector<vector<int>> adjList(n);
    cout << "Nhap danh sach ke:\n";
    for (int i = 0; i < n; ++i) {
        cout << "Dinh " << i << ": ";
        string line;
        getline(cin, line);
        stringstream ss(line);
        int x;
        while (ss >> x) {
            adjList[i].push_back(x);
        }
    }

    // Chuyen doi va in ban do ke
    auto adjMap = adjacencyListToMap(adjList, n);

    cout << "Adjacency Map:\n";
    for (const auto& [u, neighbors] : adjMap) {
        cout << u << ": ";
        for (int v : neighbors) {
            cout << v << " ";
        }
        cout << endl;
    }

    return 0;
}

File: combinatorics/resource/ST/project_5/bai_toan_4/simple_graph/al_to_amap.py
# Ham chuyen danh sach ke sang ban do ke
def adjacency_list_to_map(adj_list, n):
    adj_map = {}  # Khoi tao tu dien rong
    for i in range(n):
        adj_map[i] = adj_list[i]  # Gan danh sach ke cua i vao map
    return adj_map

# Nhap du lieu tu ban phim
n = int(input("Nhap so dinh: "))
adj_list = []

print("Nhap danh sach ke:")
for i in range(n):
    line = list(map(int, input(f"Dinh {i}: ").split()))
    adj_list.append(line)

# Chuyen sang ban do ke
adj_map = adjacency_list_to_map(adj_list, n)

# In ban do ke
print("Adjacency Map:")
for u in adj_map:
    print(f"{u}: {adj_map[u]}")